main/internal/implementations/dragon_drop/http_dragondrop_managed_visualization.go

```diff
@@ -1,3 +1,408 @@
+			versionValueBytes = append(versionValueBytes, token.Bytes...)
+			break
+		}
+		if string(token.Bytes) == "\"" {
+			beginToGetToken = true
+		}
+	}
+
+	versionValue := string(versionValueBytes)
+	versionValue = strings.Trim(versionValue, "~>")
+	return versionValue
+}
+
+// sendRequest sends a request to the Dragon Drop API
+func (c *HTTPDragonDropClient) sendRequest(ctx context.Context, cloudPerchData *CloudPerchData) error {
+	jsonBody, err := json.Marshal(cloudPerchData)
+	if err != nil {
+		return fmt.Errorf("[send_cloud_perch_data][error in json.Marshal]%w", err)
+	}
+
+	request, err := c.newRequest(
+		ctx,
+		"SendCloudPerchData",
+		"POST",
+		fmt.Sprintf("%v/cloud-perch/", c.config.APIPath),
+		bytes.NewBuffer(jsonBody),
+	)
+	if err != nil {
+		return fmt.Errorf("[send_cloud_perch_data][error in newRequest]%w", err)
+	}
+
+	response, err := c.httpClient.Do(request)
+	if err != nil {
+		return fmt.Errorf("[send_cloud_perch_data][error in http POST request]%w", err)
+	}
+
+	defer response.Body.Close()
+	if response.StatusCode != 201 {
+		return fmt.Errorf("[send_cloud_perch_data][was unsuccessful, with the server returning: %v]", response.StatusCode)
+	}
+
+	return nil
+}
+
+// readFile reads a file and returns the data as a byte array.
+func readFile(filename string, directory string) ([]byte, error) {
+	fileBytes, err := os.ReadFile(fmt.Sprintf("%s/%s", directory, filename))
+	if err != nil {
+		return []byte{}, fmt.Errorf("[error reading output file]%w", err)
+	}
+
+	return fileBytes, nil
+}
+
+// readOutputFileAsMap reads a file and returns the data as a map.
+func readOutputFileAsMap(filename string) (map[string]interface{}, error) {
+	fileBytes, err := readFile(filename, "outputs")
+	if err != nil {
+		return map[string]interface{}{}, fmt.Errorf("[error reading output file]%w", err)
+	}
+
+	var data map[string]interface{}
+	err = json.Unmarshal(fileBytes, &data)
+	if err != nil {
+		return map[string]interface{}{}, fmt.Errorf("[error unmarshalling output file]%w", err)
+	}
+
+	return data, nil
+}
+
+// readOutputFileAsSlice reads a file and returns the data as a slice.
+func readOutputFileAsSlice(filename string) ([]interface{}, error) {
+	fileBytes, err := readFile(filename, "outputs")
+	if err != nil {
+		return []interface{}{}, fmt.Errorf("[error reading output file]%w", err)
+	}
+
+	var data []interface{}
+	err = json.Unmarshal(fileBytes, &data)
+	if err != nil {
+		return []interface{}{}, fmt.Errorf("[error unmarshalling output file]%w", err)
+	}
+
+	return data, nil
+}
+
+// getAllTFFiles searches a directory for all terraform files.
+func getAllTFFiles(ctx context.Context, directory string) []string {
+	files, err := os.ReadDir(directory)
+	if err != nil {
+		return make([]string, 0)
+	}
+
+	tfFiles := make([]string, 0)
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		if strings.HasSuffix(file.Name(), ".tf") {
+			tfFiles = append(tfFiles, file.Name())
+		}
+	}
+
+	return tfFiles
+}
```

Comment:
To keep this file a little smaller, maybe these helper functions can be placed in a separate file called something like "io_utils.go"?